vortex-layout/src/layouts/chunked/stats_table.rs

```diff
@@ -146,9 +151,13 @@ impl StatsAccumulator {
     }
 
     pub fn push_chunk(&mut self, array: &Array) -> VortexResult<()> {
+        if &self.column_dtype != array.dtype() {
+            vortex_bail!("Chunk dtype does not match expected stats column dtype");
+        }
+
         for (s, builder) in self.stats.iter().zip_eq(self.builders.iter_mut()) {
             if let Some(v) = array.statistics().compute(*s) {
-                builder.append_scalar(&Scalar::new(s.dtype(array.dtype()), v))?;
+                builder.append_scalar(&Scalar::new(s.dtype(array.dtype()).as_nullable(), v))?;
```

Comment:
should Stat::dtype() function just always return a `Nullable` dtype...?

Reply:
No, this is only in the context of a stats table where a chunk encoding may not know how to compuite a stat, therefore it's missing, even though the DType of the statistic is non-nullable (i.e. matches that of the column)

Reply:
+1